htdocs/kernel/online.php

```diff
@@ -75,18 +75,15 @@ public function write($uid, $uname, $time, $module, $ip)
         $ip = $this->db->quote($ip);
 
         if ($uid > 0) {
-            $sql = 'SELECT COUNT(*) FROM ' . $this->db->prefix('online') . " WHERE online_uid={$uid}";
+            $sql = 'SELECT COUNT(*) FROM ' . $this->db->prefix('online') . " WHERE online_ip={$ip}";
         } else {
             $sql = 'SELECT COUNT(*) FROM ' . $this->db->prefix('online')
                    . " WHERE online_uid={$uid} AND online_ip={$ip}";
         }
         list($count) = $this->db->fetchRow($this->db->queryF($sql));
         if ($count > 0) {
             $sql = 'UPDATE ' . $this->db->prefix('online')
-                   . " SET online_updated = {$time}, online_module = {$module} WHERE online_uid = {$uid}";
-            if ($uid === 0) {
-                $sql .= " AND online_ip={$ip}";
-            }
+                   . " SET online_updated = {$time}, online_module = {$module}, online_uid = {$uid}  WHERE online_ip={$ip}";
```

Comment:
This ignores the possibility of multiple users sharing the same outward IP. Simple example, everyone that uses my wifi gets the same public IP. Most companies of any size will have proxy servers that will do the same, 100 people sharing a small pool of actual IP addresses.

I think if you added one more condition in the WHERE clause, it might be better:
`WHERE online_ip={$ip} AND online_uid IN (0, {$uid})`

Reply:
Thank you Richard, it's much better!